Joss paper

vyasr opened the pull request:
This PR adds a draft of a paper for publication in JOSS. I have currently marked it as a draft because one or two of the statements are aspirational. In particular, we want most of the properties to be settable, but in practice many of them are not yet. @Tobias-Dwyer has put together a nice feature matrix for us to work off of, and I think we should aim for higher coverage of setting before considering this paper ready. Many of the features should be quite simple to implement (e.g. rescaling the in/circumsphere radii), so I suggest we progress through that list. I don't think it will be too difficult to establish a natural stopping point based on which items prove either unnecessary or too difficult to prioritize.

Reviewers should feel free to make changes directly as they see fit, but for larger changes please comment first to make sure everyone is on the same page and to avoid any conflicting changes. Some thoughts:

- Some more recent papers include explicit listings of features. I'm not inclined to include something like that, but I'm open to other opinions.
- JOSS claims a 250-1000 word limit, but I looked at recent papers and many of them seem to blatantly violate the limit, so I'm not sure to what degree it's enforced. We're currently inside the limits, but in any case I'd recommend revisiting closer to the end.
- I'm a fan of figures, but I can't think of any that are both relevant and useful enough to not seem superfluous. If anyone has ideas feel free to contribute them.

Here is some material you might find useful in reviewing this pull request.

- Use [this link](https://whedon.theoj.org/) to compile the paper (you'll have to specify the repository URL and the "joss_paper" branch) and see how it looks.
- Here are [JOSS's guidelines](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) for the paper structure.
- Here's the [main page](https://joss.theoj.org/) where you can see recently published papers for examples.

bdice reviewed (commented):
Nice work @vyasr. I have comments attached. I think there are opportunities for improvement in a few areas:
- Some parts of the paper are repetitive. For example, some information about shape classes and shape families is given before the Statement of Need and then again in the Statement of Need. I'd rather see that information combined into one concise statement. In order to make it flow better, it might be good to put the Statement of Need much higher. The first few paragraphs already try to justify the purpose/focus/existence of the package. I propose 1 short paragraph for "Summary," then 2 paragraphs for "Statement of Need," then 2-3 paragraphs separated into headings like "Package Features" and "Applications."
- This paper doesn't discuss the conceptual design (abstract classes for 2D/3D shapes) or other scientific software (usage of NumPy/SciPy/rowan, interfaces with GSD shape specs, inline plotting via matplotlib / Jupyter)
- This skipped over sphero-shapes, which is a significant motivation for the package because it's relatively hard to compute properties of them.
- This should include some kind of table or figure. A table of features would be informative but somewhat boring. I propose a picture of the polyhedra zoo family, or a picture of every presently supported shape class.

Good for a first draft. Eager to work with you on the revisions and/or discuss further on a call.

vyasr commented:
@bdice thanks for the feedback. Would you like to get on a call (perhaps also with @Tobias-Dwyer) and hash some of these things out? There's enough to discuss that I do think that would be easier. At present I don't even necessarily want to accept many of your suggestions, since they generally suggest a need for broader reorganization and making small local changes would just obscure that.

Tobias-Dwyer commented:
I'd be down for a call to discuss @bdice's comments and what we think still needs to be done to get a plan in order to complete the paper. Just send me the time and zoom link :D

vyasr commented on paper/paper.md:
```diff
@@ -0,0 +1,77 @@
+
+Considerations of shape are becoming increasingly important in materials science as improved synthetic capabilities have allowed the creation of a wide range of anisotropic particles [@Glotzer2007b].
+Colloidal science in particular has seen immense growth in this area, and numerous studies have shown that particle shape is an important handle for controlling the self-assembly of colloidal crystals [@Damasceno2012d,@Glotzer2007b,@Chen2014].
+Precise modeling of these systems requires robust, reproducible methods for generating shapes and calculating their properties [@Anderson2020,@Ramasubramani2020b,@Allen2006], a gap that coxeter aims to fill.
```
I think HOOMD and the Allen paper are appropriate citations for this particular sentence, since they are examples of why we need robust calculations of these properties: for the types of dynamics algorithms encoded by tools like this. However, they're also basically two random examples out of hundreds of possibly valid citations, so maybe we should just remove them since it's a pretty obvious statement?

Regarding papers that actually use coxeter/euclid: the citation of my ALJ paper is actually relevant of a project that used this, and since it's the only published work I'm aware of that has used coxeter since its rewrite/conversion from euclid, it's probably the best citation that we have. From a quick search for `import euclid` in our nfs space (only up to very limited depth, so I'm sure I missed scripts that are deeply nested somewhere in the directory structure), projects using euclid or coxeter are:

- All of Bryan's, so I think we can safely cite any of his published papers.
- Rosy's TetragonalDiamond project, both in the photonics code and in PMFT calculations. Since the PMFTs are in her paper, I think we can probably cite it (the Phys Rev Mat one).
- Kelly's dodecagonal quasicrystal project, which has currently been submitted. I don't know what the status is, but it's possible that it'll get published by the time that this paper is ready, in which case we could include it.
- Yein's bipyramid work, which I have no idea what state it's in.
- Sangmin's two step nucleation project, which is published but his usage of euclid seems super minimal so without asking him for more details I wouldn't want to cite it as an example.


The other question is where discussion of papers using coxeter fits best in the new draft.

vyasr replied:
@bdice any more thoughts on this front?

vyasr commented on paper/paper.md:
```diff
@@ -0,0 +1,77 @@
+
+# Acknowledgements
+
+**Update acknowledgments**
```
Yep I'll update this whole thing at the end. Right now it's just people who I don't want to forget, I'm not worried about funding acknowledgments.

vyasr commented on paper/paper.md:
```diff
@@ -0,0 +1,77 @@
+Most computational geometry libraries typically focus on solving more complex problems like finding convex hulls, Voronoi tesselations, and Delaunay triangulations [@cgal].
+The purpose of coxeter is to provide a standard implementation for simpler calculations such as volumes, bounding spheres, and inertia tensors, for which formulas are generally well-known but often require careful consideration to implement robust and efficient algorithms to compute.
+Inertia tensors are a particularly notable example since they can be quite difficult to compute accurately for arbitrary polytopes and require specialized algorithms [@Kallay2006].
+Physics-based models, for instance molecular dynamics simulations based on fundamental laws of motion, can be highly sensitive to the precise values of these quantities, necessitating their accurate evaluation.
```
Rather than a specific implementation section, I've interspersed these citations where I think they're most appropriate. Let me know if you think there's a better way to do this.

vyasr commented on paper/paper.md:
```diff
@@ -0,0 +1,77 @@
+The library also serves as a repository for the generation of shapes.
+While simple classes of shapes like spheres and ellipsoids can be described via a small fixed set of parameters, the definitions of polygons and polyhedra can be arbitrarily long depending on the number of vertices of these shapes.
+The shape family API in coxeter provides a flexible way to define and work with collections of related shapes, ranging from enumerable sets like the Platonic solids to continuously defined sets of shapes [@Chen2014].
+These different types of shape families are handled using identical APIs, so users can easily switch between shapes that have completely different mathematical definitions using a single line of code.
```
I think I missed this one, and given the current stage of the draft I'm not sure what to do with this. Have a read through the current draft then let's see how we feel?

vyasr commented:
@Tobias-Dwyer @bdice I've updated the draft. I tried to address most of what we discussed in our review. I didn't apply any of the inline suggestions directly since so much was completely rewritten, but I did go back through after the fact to verify that any issues raised in the first round of review were addressed in this revision if that part of the text hadn't changed. I left some of those conversations unresolved where I thought there was additional discussion to be had and added to that part of the conversation.

Let's plan to meet again once you two have had a chance to read through the revision.

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+
+# Package Overview
+
+The coxeter Python package provides the tools to represent, generate, and compute properties of shapes in two and three dimensions.
```
```suggestion
The coxeter Python package provides tools to represent, generate, and compute properties of shapes in two and three dimensions.
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+# Package Overview
+
+The coxeter Python package provides the tools to represent, generate, and compute properties of shapes in two and three dimensions.
+The package emphasizes simplicity and flexibility, using a common set of abstractions to present a largely uniform interface across various shapes and allowing easy mutation of almost all of their geometric attributes.
```
This is an _excellent_ sentence.

vyasr replied:
Thanks :)

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+
+The coxeter Python package provides the tools to represent, generate, and compute properties of shapes in two and three dimensions.
+The package emphasizes simplicity and flexibility, using a common set of abstractions to present a largely uniform interface across various shapes and allowing easy mutation of almost all of their geometric attributes.
+The package also aims to function as a repository for specific groups of shapes, exposing an easy-to-use API for shape generation that users can easily extend to make particular sets of shapes collectively accessible.
```
- "aims to function" is too verbose. Replaced with "serves"
- "easily" is redundant with "easy-to-use"
- "specific groups of shapes" is redundant with "particular sets of shapes." I suggested "particular geometric objects" but that may be less precise.
```suggestion
The package also serves as a repository for specific groups of shapes, exposing an easy-to-use API for shape generation that users can extend to make particular geometric objects collectively accessible.
```

vyasr replied:
First two changes are definitely good. I think the third is an improvement, maybe not what we want in the final version but we can reassess in future drafts.

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+
+# Statement of Need
+
+Considerations of shape are becoming increasingly important in materials science as improved synthetic capabilities have allowed the creation of a wide range of anisotropic particles [@Glotzer2007b].
+Colloidal science in particular has seen immense growth in this area, and numerous studies have shown that particle shape is an important handle for controlling the self-assembly of colloidal crystals [@Damasceno2012d,@Glotzer2007b,@Chen2014].
```
I think we need to mention something like "colloidal assembly" or "crystals of nanoparticles" earlier because otherwise there's no clear connection between "shape" and "materials science." I want to swap some bits between this sentence and the following sentence. Here's my proposal:
```suggestion
Materials science has recently seen advances in nanoparticle synthesis, allowing the creation of a wide range of anisotropic (non-spherical) particles [@Glotzer2007b].
Consequently, the idea of particle shape in colloidal and nanoscale assembly has seen immense growth.
Numerous studies have shown that particle shape is an important handle for controlling the self-assembly of colloidal crystals [@Damasceno2012d,@Glotzer2007b,@Chen2014].
```

vyasr replied:
I'm not sure that your changes are addressing your concern. When you say earlier, do you mean closer to the beginning of the SOP? Although in your new version you've shortened the first sentence a bit, the content hasn't really changed, so the same problem persists, unless you really are concerned that the second sentence in the original is too long and so mention of self-assembly at the end is "too late". Perhaps it would be better to swap "anisotropic particles" for "particles with nontrivial shapes" in the first sentence, or something similar to indicate the shape aspect?

vyasr replied:
@bdice thoughts?

bdice replied:
I'm resolving this with no action. It's fine as it is.

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+Considerations of shape are becoming increasingly important in materials science as improved synthetic capabilities have allowed the creation of a wide range of anisotropic particles [@Glotzer2007b].
+Colloidal science in particular has seen immense growth in this area, and numerous studies have shown that particle shape is an important handle for controlling the self-assembly of colloidal crystals [@Damasceno2012d,@Glotzer2007b,@Chen2014].
+Precise modeling of these systems requires reproducible methods for generating shapes and calculating their properties [@Anderson2020,@Allen2006] **Add more citations of real shapes**.
+An important aspect of achieving this reproducibility is making canonical definitions of shapes used in particular studies readibly available to other researchers.
```
```suggestion
An important aspect of achieving this reproducibility is making canonical definitions of shapes used in particular studies readily available to other researchers.
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+
+# Summary of Features
+
+## Shape Analysis **better title?**
```
Computing Geometric and Physical Properties of Shapes? It's a long subsection title but it summarizes the intent.

vyasr replied:
If we drop "of Shapes" then that seems good: "Computing Geometric and Physical Properties".

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+## Shape Analysis **better title?**
+
+The central elements in coxeter are the shape classes, which encode the features of particular types of shapes.
+In order to enforce a highly uniform API and ensure conceptual clarity, all shape classes extend a small set of abstract base classes that encode specific subsets of properties: for instance, the standard properties of all shapes in two dimensions are embedded in the ``Shape2D`` class.
```
```suggestion
In order to enforce a highly uniform API and ensure conceptual clarity, all shape classes inherit from a small set of abstract base classes that encode specific subsets of properties: for instance, the standard properties of all shapes in two dimensions are embedded in the ``Shape2D`` class.
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+
+The central elements in coxeter are the shape classes, which encode the features of particular types of shapes.
+In order to enforce a highly uniform API and ensure conceptual clarity, all shape classes extend a small set of abstract base classes that encode specific subsets of properties: for instance, the standard properties of all shapes in two dimensions are embedded in the ``Shape2D`` class.
+In addition to standard types of shapes such as ellipsoids or polygons, coxeter also includes more esoteric shape classes like spheropolyhedra, which are important to modeling the imperfect rounded polyhedra frequently synthesized at the nano- and micron scales.
```
```suggestion
In addition to standard types of shapes such as ellipsoids or polygons, coxeter also includes more esoteric shape classes like spheropolyhedra, which are important for modeling the imperfect rounded polyhedra frequently synthesized at the nano- and micron scales.
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+The central elements in coxeter are the shape classes, which encode the features of particular types of shapes.
+In order to enforce a highly uniform API and ensure conceptual clarity, all shape classes extend a small set of abstract base classes that encode specific subsets of properties: for instance, the standard properties of all shapes in two dimensions are embedded in the ``Shape2D`` class.
+In addition to standard types of shapes such as ellipsoids or polygons, coxeter also includes more esoteric shape classes like spheropolyhedra, which are important to modeling the imperfect rounded polyhedra frequently synthesized at the nano- and micron scales.
+Even simple properties like surface areas are generally nontrivial to compute for such shapes, but using coxeter they are no more difficult to work with than other, more common, shapes.
```
Common is hard to define. Also I lost the antecedent "spheropolyhedra" somewhere. Let's restate it for clarity.
```suggestion
Even simple properties like surface areas are generally nontrivial to compute for such shapes, but using coxeter spheropolyhedra are no more difficult to work with than any other 3D shape.
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+
+The shape classes transparently expose many geometric attributes in the form of settable Python properties, allowing on-the-fly rescaling or transformation of the shape.
+This aspect of coxeter is designed to fill a common gap in most computational geometry libraries, which typically focus on solving more complex problems like finding convex hulls, Voronoi tessellations, and Delaunay triangulations [@cgal]; coxeter aims to provide a standard implementation for simpler calculations such as surface areas and bounding spheres for which formulas are generally well-kno ...
+These properties range from standard calculations like volumes and surface areas to less common metrics like mean curvatures and asphericities that are relevant for specific research avenues [@Irrgang2017].
```
I'd insert a concrete phrase instead of "specific research avenues," it sounds really esoteric and doesn't indicate the scientific utility of this package. e.g. "Understanding how shape properties correspond to self-assembly behavior of colloidal crystals."

vyasr replied:
How about "computing equations of state for nonspherical particles"?

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+In addition to purely geometric properties, shapes in coxeter also expose various physically relevant quantities in order to support a wide range of applications.
+Some examples of such properties are inertia tensors, which are integral to the equations of motion for anisotropic bodies, and scattering form factors, which are essentially Fourier transforms of the shape volume that are critical to characterizing structure in condensed matter physics.
+Since physical equations and observables can be highly sensitive to inputs like inertia tensors, coxeter emphasizes robust methods for their evaluation [@Kallay2006].
+Two dimensional shapes like polygons are embedded in three dimensions rather than in the plane, so coxeter uses the rowan library [@Ramasubramani2018] to rotate them into the plane and then compute various properties to avoid complications and numerical instabilitiesthat arise from performing integrals over planar lamina embedded in $\mathcal{R}^3$.
```
If you disagree with my second edit and wish to use R^3 instead of "3D Euclidean space," I think we want `\mathbb{R}^3` over `\mathcal{R}^3`.
```suggestion
Two dimensional shapes like polygons are embedded in three dimensions rather than in the plane, so coxeter uses the rowan library [@Ramasubramani2018] to rotate them into the plane and then compute various properties to avoid complications and numerical instabilities that arise from performing integrals over planar lamina embedded in 3D Euclidean space.
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+These different types of shape families are handled using identical APIs, so users can easily switch between shapes that have completely different mathematical definitions using a single line of code.
+Shape families generate coxeter shape classes from input parameters, simplifying access to computed geometric and physical properties.
+
+A number of such families are bundled into coxeter, but just as importantly, the framework allows users to work with arbitrary lists of shapes provided as dictionaries of attributes corresponding to specified schemata.
```
...schemata? Too strange of a word. I don't even know what the end of the sentence means, so I deleted it.
```suggestion
A number of such families are bundled into coxeter, but just as importantly, the framework allows users to work with arbitrary lists of shapes provided as dictionaries of attributes.
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,93 @@
+This dictionary-based definition can be trivially converted to JSON, making it trivial to share representations of shapes.
+The library also stores mappings from digital object identifiers (DOIs) to families, so that any user can contribute families associated with published research to make them immediately collectively accessible.
+We anticipate that the set of shape families in coxeter will grow over time as users generate and contribute their shape families to coxeter, with the goal of providing a centralized repository for use in reproducing and extending prior research, particularly in the field of shape-driven nanoparticle self-assembly.
+Currently coxeter primarily supports the schema proposed by the GSD library **should we cite? there's no canonical source other than RTD**, making it directly compatible with the HOOMD-blue molecular simulation tool [@Anderson2020], but other schema can easily be implemented on an as-needed basis.
```
Yes, we should cite GSD as if it were a web page, and point to the GitHub source.

bdice reviewed (commented):
This is a major step forward. I really like the reorganization! A few minor comments in this review. The only major work I see left is generating a figure and filling any other gaps that we identified in our last discussion.

vyasr commented:
@bdice thanks for the review! There are a couple of comments left from your previous review that require a little discussion, and I've responded to the ones in your most recent review that require further commentary.

@Tobias-Dwyer how do you feel about this draft?

Tobias-Dwyer commented:
So I like this draft a lot more. It more simply addresses why a user would be interested in this package and what calculations and functionality it brings to the table over other software. I made some slight wording changes. Also, I don't know if we want to emphasize any limitations of the package, for example all physical properties we calculate are at constant density. I assume that would matter somewhat if you were looking at using coxeter to calculate particle properties and had a core-shell particle or if we had a shape made of 2 different materials. I sort of address this in a wording change above, but idk if we want to go into things that package isn't really made to do conveniently? Otherwise, I think the draft is more clearly worded.

vyasr commented:
This paper still needs acknowledgments updated, but aside from that I think it's ready for final rounds of review.

bdice commented on paper/paper.bib:
This is formatted weirdly.
```suggestion
@article{Kao2019,
author ="Kao, Peng-Kai and VanSaders, Bryan J. and Durkin, Michael D. and Glotzer, Sharon C. and Solomon, Michael J.",
title = "Anisotropy effects on the kinetics of colloidal crystallization and melting: comparison of spheres and ellipsoids",
journal = "Soft Matter",
year = "2019",
volume = "15",
issue = "37",
pages = "7479-7489",
publisher = "The Royal Society of Chemistry",
doi = "10.1039/C9SM00887J",
abstract = "We use alternating current (AC) electric field assisted self-assembly to produce two-dimensional{,} millimeter scale arrays of ellipsoidal colloids and study the kinetics of their phase reconfiguration by means of confocal microscopy{,} light scattering{,} and computer simulation. We find that the kinetics of orientational and positional ordering can be manipulated by changing the shape of the colloids: ellipsoids with aspect ratio 2.0 melt into disordered structures 5.7 times faster compared to spheres. On the other hand{,} ellipsoids self-assemble into ordered crystals at a similar rate to spheres. Confocal microscopy is used to directly visualize defects in the self-assembled structures. Small-angle light scattering (SALS) quantifies the light diffraction response{,} which is sensitive to the kinetics of positional and orientational ordering in the self-assembled anisotropic structures. We find three different light diffraction patterns: a phase with high orientational order (with chain-like structure in real space){,} a phase with high positional and orientational order (characteristic of a close-packed structure){,} and a phase that is disordered in position but with intermediate orientational order. The large influence of aspect ratio on the kinetics of the positionally and orientationally ordered phase is explored through simulation; it is found that the number of particle degrees of freedom controls the difference between the melting rates of the ellipsoids and spheres. This research contributes to the understanding of reconfiguration kinetics and optical properties of colloidal crystals produced from anisotropic colloids."}
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,96 @@
+  - geometry
+  - physics
+  - materials science
+authors:
+  - name: Vyas Ramasubramani
+    orcid: 0000-0001-5181-9532
+    affiliation: 1
+  - name: Bradley Dice
+    orcid: 0000-0002-9983-0770
+    affiliation: 3
+  - name: Tobias Dwyer
+    orcid: 0000-0001-6443-7744
+    affiliation: 1
+  - name: Sharon C. Glotzer
+    orcid: 0000-0002-7197-0085
+    affiliation: "1, 2, 3, 4"
+affiliations:
+ - name: Department of Chemical Engineering, University of Michigan
+   index: 1
+ - name: Department of Materials Science and Engineering, University of Michigan
+   index: 2
+ - name: Department of Physics, University of Michigan
+   index: 3
+ - name: Biointerfaces Institute, University of Michigan
+   index: 4
```
This needs to be re-ordered so that the numbers are "Ramasubramani (1), Dice (2), Dwyer (1), Glotzer (1, 2, 3, 4)."
```suggestion
authors:
  - name: Vyas Ramasubramani
    orcid: 0000-0001-5181-9532
    affiliation: 1
  - name: Bradley Dice
    orcid: 0000-0002-9983-0770
    affiliation: 2
  - name: Tobias Dwyer
    orcid: 0000-0001-6443-7744
    affiliation: 1
  - name: Sharon C. Glotzer
    orcid: 0000-0002-7197-0085
    affiliation: "1, 2, 3, 4"
affiliations:
 - name: Department of Chemical Engineering, University of Michigan
   index: 1
 - name: Department of Physics, University of Michigan
   index: 2
 - name: Department of Materials Science and Engineering, University of Michigan
   index: 3
 - name: Biointerfaces Institute, University of Michigan
   index: 4
```

bdice commented on paper/paper.md:
```diff
@@ -0,0 +1,96 @@
+This dictionary-based definition can be simply converted to JSON, making it trivial to share representations of shapes.
+The library also stores mappings from digital object identifiers (DOIs) to families, so that any user can contribute families associated with published research to make them immediately collectively accessible.
+We anticipate that the set of shape families in coxeter will grow over time as users generate and contribute their shape families to coxeter, with the goal of providing a centralized repository for use in reproducing and extending prior research, particularly in the field of shape-driven nanoparticle self-assembly.
+Currently coxeter primarily supports the schema proposed by the GSD library [@GlotzerLabGSD], making it directly compatible with the HOOMD-blue molecular simulation tool [@Anderson2020], but other schema can easily be implemented on an as-needed basis.
```
```suggestion
Currently coxeter primarily supports the schema proposed by the GSD library [@GlotzerLabGSD], making it directly compatible with the HOOMD-blue molecular simulation tool [@Anderson2020], but other schema can be implemented as needed.
```

bdice reviewed (approved):
@vyasr @Tobias-Dwyer This looks great! I am very happy with the current state of the paper. I left a couple small comments, and the following reminders:

- Fix this citation so it says "Glotzer Lab" like how "The CGAL Project" renders:
`Lab, G. (2021).GSD v2.4.0.https://github.com/glotzerlab/gsd`
- Finalize acknowledgements
- Update the date when submitting

bdice commented:
@vyasr @Tobias-Dwyer I reviewed and updated all the citations. It turns out that names are very tricky, especially last names with leading words like "van." Everything in the references looks right to me now. This should be ready for a final approval from each of you, I think it's good to go.

vyasr commented:
@Tobias-Dwyer @bdice one final note re: acknowledgments. I would also like to acknowledge the authors of our external modules. Especially if we're going to acknowledge all contributors, we should also mention the three external repos that we're embedding. Are you two cool with that? If so it might also be nice to reach out and let them know.

bdice commented:
@vyasr Yes, that sounds fine to me!
